### PR TITLE
API: Endpoint to get an organization's details

### PR DESCRIPTION
**Pull request by twinsfan421**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This would provide an api show endpoint `dev.to/api/organizations/{org_username}` to search organizations by their username.  
As discussed in issue 9212, This is the first of 4 related endpoints that will include:
```
dev.to/api/organizations/{org_username}/users
dev.to/api/organizations/{org_username}/articles
dev.to/api/organizations/{org_username}/listings
```
## Related Tickets & Documents
https://github.com/forem/forem/issues/9212

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.forem.com
- [ ] readme
- [ ] no documentation needed

### CODE REVIEW

**Review by mstruve — COMMENTED** [review]
Hey @twinsfan421! Code looks super solid here. Could you also add some API docs for this new endpoint? `docs/api_v0.yml` 

Please and thank you!!!

**Review by citizen428 — APPROVED** [review]

**Comment by twinsfan421** [comment]
> Hey @twinsfan421! Code looks super solid here. Could you also add some API docs for this new endpoint? `docs/api_v0.yml`
> 
> Please and thank you!!!
> 
> PS also looks like there is a routes.rb conflict :)

Ok thanks, will do!

**Review by benhalpern — CHANGES_REQUESTED** [review]
Thanks for the PR. This should include an update to the API docs, as per described here. https://docs.forem.com/contributing/api/

Here's an example of the last PR with API changes that include the docs. https://github.com/forem/forem/pull/10547

**Comment by twinsfan421** [comment]
I pushed up a fix for the merge conflict and an update to the api docs.  Let me know if you want me to make any changes.

File: app/views/api/v0/organizations/show.json.jbuilder [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  :twitter_username,
+  :github_username,
+  :url,
+  :location,
+  :tech_stack,
+  :tag_line,
+  :story,
+)
+
+json.joined_at @organization.created_at.strftime("%b %e, %Y")
```
Comment (rhymes) [inline app/views/api/v0/organizations/show.json.jbuilder:17]: This as all the other dates should be in ISO8601/RFC3339 format as specified by the spec. Unfortunately `joined_at` being a "human date" in the user object is a remnant of the past.

**Review by rhymes — CHANGES_REQUESTED** [review]
Thanks @twinsfan421, great work! I would align the joined/creation date to the format of the other dates

File: app/views/api/v0/organizations/show.json.jbuilder (hunk 1 above)
Comment (twinsfan421) [reply app/views/api/v0/organizations/show.json.jbuilder]: Thanks @rhymes for reviewing. I was following the user show endpoint but just uploaded a change to the iso format.

File: docs/api_v0.yml [hunk 2]
```diff
@@ -1104,6 +1104,48 @@ components:
...
+          type: string
+          nullable: true
+        tag_line:
+          type: string
+          nullable: true
+        story:
+          type: string
+          nullable: true
+        joined_at:
+          description: Date of joining (formatted with strftime `"%b %e, %Y"`)
```
Comment (rhymes) [inline docs/api_v0.yml:1142]: I think we can remove the part in the parenthesis here and then merge :D , also can you add `format: date-time` to the spec of this field ?

**Review by rhymes — COMMENTED** [review]
Thanks a lot for the changes @twinsfan421, one tiny last bit left :)

File: docs/api_v0.yml (hunk 2 above)
Comment (twinsfan421) [reply docs/api_v0.yml]: No problem. I just pushed up the changes for that. Thanks

**Review by rhymes — APPROVED** [review]
Thanks @twinsfan421, LGTM!

**Review by benhalpern — APPROVED** [review]

**Pull request merged**